winpkg/ddev_windows_installer.nsi

```diff
@@ -0,0 +1,1967 @@
+!include "MUI2.nsh"
+!include "LogicLib.nsh"
+!include "WinMessages.nsh"
+!include "FileFunc.nsh"
+!include "Sections.nsh"
+!include "x64.nsh"
+!include "WordFunc.nsh"
+!include "StrFunc.nsh"
+
+!insertmacro GetParameters
+!insertmacro GetOptions
+
+!insertmacro WordFind
+${StrStr}
+; Remove the Trim macro since we're using our own TrimWhitespace function
+
+!ifndef TARGET_ARCH # passed on command-line
+  !error "TARGET_ARCH define is missing!"
+!endif
+
+Name "DDEV"
+OutFile "..\.gotmp\bin\windows_${TARGET_ARCH}\ddev_windows_${TARGET_ARCH}_installer.exe"
+
+; Use proper Program Files directory for 64-bit applications
+InstallDir "$PROGRAMFILES64\DDEV"
+RequestExecutionLevel admin
+
+!define PRODUCT_NAME "DDEV"
+!define PRODUCT_VERSION "${VERSION}"
+!define PRODUCT_PUBLISHER "DDEV Foundation"
+
+; Variables
+Var /GLOBAL INSTALL_OPTION
+Var /GLOBAL SELECTED_DISTRO
+Var /GLOBAL SILENT_INSTALL_TYPE
+Var /GLOBAL SILENT_DISTRO
+Var /GLOBAL WINDOWS_CAROOT
+Var /GLOBAL DEBUG_LOG_HANDLE
+Var /GLOBAL DEBUG_LOG_PATH
+Var StartMenuGroup
+
+!define REG_INSTDIR_ROOT "HKLM"
+!define REG_INSTDIR_KEY "Software\Microsoft\Windows\CurrentVersion\App Paths\ddev.exe"
+!define REG_UNINST_ROOT "HKLM"
+!define REG_UNINST_KEY "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}"
+
+; Installer Types
+InstType "Full"
+InstType "Simple"
+InstType "Minimal"
+
+!define MUI_ICON "graphics\ddev-install.ico"
+!define MUI_UNICON "graphics\ddev-uninstall.ico"
+
+!define MUI_INSTFILESPAGE_ABORTHEADER_TEXT "Installation Aborted"
+!define MUI_INSTFILESPAGE_ABORTHEADER_SUBTEXT "Setup was not completed successfully."
+
+!define MUI_HEADERIMAGE
+!define MUI_HEADERIMAGE_BITMAP "graphics\ddev-header.bmp"
+!define MUI_WELCOMEFINISHPAGE_BITMAP "graphics\ddev-wizard.bmp"
+
+!define MUI_ABORTWARNING
+
+; Function declarations - must be before page definitions
+
+; InitializeDebugLog - Open debug log file for writing
+Function InitializeDebugLog
+    StrCpy $DEBUG_LOG_PATH "$TEMP\ddev_installer_debug.log"
```

Comment:
I can't find this file `ddev_installer_debug.log` inside `C:\Windows\Temp`, should it be created all the time?

What is `$TEMP` ?

Reply:
When a failure happens, it reports it. But 
```
PS C:\Windows> $env:TEMP
C:\Users\randy\AppData\Local\Temp
```